src/Compilers/Core/Portable/CommandLine/SarifV2ErrorLogger.cs

```diff
@@ -67,6 +67,20 @@ public override void LogDiagnostic(Diagnostic diagnostic)
                 _writer.WriteArrayStart("suppressions");
                 _writer.WriteObjectStart(); // suppression
                 _writer.Write("kind", "inSource");
+
+                if (Compilation != null)
+                {
+                    object? justification = diagnostic.GetSuppressionInfo(Compilation)?.Attribute
+                                                                        ?.NamedArguments
+                                                                        .FirstOrDefault(o => o.Key == "Justification")
+                                                                        .Value
+                                                                        .Value;
+                    if (justification != null)
+                    {
+                        _writer.Write("justification", (string)justification);
```

Comment:
```suggestion
                    if (justification is string justificationStr)
                    {
                        _writer.Write("justification", justificationStr);
```

The cast to `string` can fail at runtime, can avoid this by changing the `null` check